### PR TITLE
Add Azure Feature AzureFileShare

### PR DESCRIPTION
**Pull request by adityagesh**
Please check commits for changes

### CODE REVIEW

**Comment by adityagesh** [comment]
Started PR for early review comments

File: lisa/sut_orchestrator/azure/features.py [hunk 1]
```diff
@@ -2757,3 +2760,140 @@ def reset_password(self, username: str, password: str) -> None:
...
+                subscription_id=platform.subscription_id,
+                cloud=platform.cloud,
+                account_name=storage_account_name,
+                file_share_name=share_name,
+                resource_group_name=resource_group_name,
+                log=self._log,
+            )
+        return fs_url_dict
+
+    """
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: Move it into the method.

Comment (adityagesh) [reply lisa/sut_orchestrator/azure/features.py]: Updated

File: microsoft/testsuites/core/storage.py [hunk 2]
```diff
@@ -625,3 +693,17 @@ def _check_root_partition_in_log(
         ):
             return False
         return True
+
+    def _reload_fstab_config(self, node: Node) -> None:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/storage.py:719]: move it to mount tool.

Comment (adityagesh) [reply microsoft/testsuites/core/storage.py:719]: Updated

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Comment by LiliDeng** [comment]
@adityagesh please fix the CI checker error.

**Comment by adityagesh** [comment]
> @adityagesh please fix the CI checker error.

done

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**